README.md

```diff
@@ -197,3 +197,90 @@ This isn't COMP2041, we don't care if you miss a couple links, just make sure yo
+HTTP/1.0 200 Connection established
+
+HTTP/1.1 200 OK
+Content-Length: 66
+Content-Type: text/plain
+Date: Mon, 01 Jun 2020 10:14:22 GMT
+Server: whoami
+X-Ctfproxy-Trace-Context: 00000000-0000-0000-0000-0000000000
+Connection: close
+
+Hello @todo! You are authenticated as todo@quoccabank.com.
+```
+
+Note that we needed to use the `-k` flag to allow insecure certificates. If you really want to avoid having to specify this option, you can add Burp's certificate as a system trusted certificate, this is probably best done at your own risk in a dedicated environment (if there's an alternative I would love to know - @todo). In order to do this you first need to export Burp's certificate (Burp > Proxy > Options > Import/Export Certificate > Export.Certificate in DER Format). You can then convert this to a pem and add it to your system's ca-certificates. _This was tested on Ubuntu @ WSL2_
+
+```bash
+% openssl x509 -in exported_cert.der -inform DER -out portswigger.crt -outform PEM
+% mkdir /usr/local/share/ca-certificates/comp6443
+% mv portswigger.crt /usr/local/share/ca-certificates/comp6443
+% update-ca-certificates
+```
+
+If you want to do this more generally you could also install and configure a tool like proxychains (proxychains-ng).
+
+<h5 class="hide-from-toc">Optional: Setting up Burp</h5>
+
+Personally I (@todo) like to have automated tools and manual browsing to go through different Burp proxies. This allows you to filter your proxy history to only contain manual browsing. We can add a new proxy as follows:
+
+1. Go to Burp > Proxy > Options
+2. In the fieldset Proxy Listeners click Add
+3. Add a new port to listen on in the "Bind to port" field and select the address you would like the proxy to bind to (loopback only is usually fine, if you're running WSL[2] or using a VM you may have to set a specifc address)
+4. Use this address and port instead of the default Burp proxy for your automated tooling.
+5. You can then navigate to your HTTP history, click on the filter bar up the top, and filter based on proxy port.
+
+<h4 class="hide-from-toc">Extracting Certificates from the PKCS#12 archive</h4>
+
+If you are writing your own scripts or don't want to proxy your traffic through burp, you can also extract the certificate and key from within the PKCS#12 file. You will need openssl installed and your installation key on hand.
+
+```bash
+$ openssl pkcs12 -in 6443.p12 -nodes
+```
+
+This will ask you for your import password, and print the certificate and key to stdout. Copy the content of the certificate and key (beginning with `-----BEGIN XXX -----` (inclusive)) and save it as two separate files.
+
+Once you have these two files, you can use them in your scripts. For example, a simple nc style connection might look like
+
+```bash
+$ openssl s_client -quiet -ign_eof -cert ~/certs/6443.pem -key ~/certs/6443.key -connect whoami.quoccabank.com:443
+```
+
+Or you can use the certificates in python requests
```

Comment:
Since we did curl above, perhaps add that you can also use curl with the certs? e.g. curl https://whoami.quoccabank.com -k --cert client.pem --key client.key

Reply:
Good Idea. Added